compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -187,6 +187,14 @@ pub(super) fn builtin(
         BuiltinLintDiagnostics::UnexpectedCfgName((name, name_span), value) => {
```

Comment:
Be aware that `value` here can be `Some(..)`, your logic should probably not trigger if it `Some(...)`.

Reply:
@chenyukang To be more precise `UnexpectedCfgName` can and is fired for things like `#[cfg(target_osss = "linux")]`. So when `value.is_some()` we should not try to suggest using the name as a value since a value is already provided.

Your logic should become something like this:
```rust
    let names_possibilities = if value.is_none() {
        ...
    } else {
        None
    };
```

Please also add a test for this, `#[cfg(linux = linux)]` or something else.

Reply:
ah, fixed it, thanks!